### PR TITLE
Added Services

### PR DESCRIPTION
**Pull request by isfandiyoro5**
Added Services, 
async await 
Added Exceptions

### CODE REVIEW

File: TestHotel/TestHotel.Service/Service/Services/EmployeeService.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+using TestHotel.DataAccess.Repository.Repositories;
+using TestHotel.Service.Service.IServices;
+
+namespace TestHotel.Service.Service.Services
+{
+    internal class EmployeeService: IEmployeeService
+    {
+        private readonly IEmployeeRepository _employeeRepository;
+
+        public EmployeeService(EmployeeRepository employeerepository)
```
Comment (mrIminjonov) [inline TestHotel/TestHotel.Service/Service/Services/EmployeeService.cs]: employeerepository ni employeeRepository qibqoy

File: TestHotel/TestHotel.Service/Service/Services/EmployeeService.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            _employeeRepository= employeerepository;
```
Comment (mrIminjonov) [inline TestHotel/TestHotel.Service/Service/Services/EmployeeService.cs]: bu yerda ham

**Review by mrIminjonov — CHANGES_REQUESTED** [review]

File: TestHotel/TestHotel.Service/Service/IServices/IBillService.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace TestHotel.Service.Service.IServices
+{
+    internal interface IBillService
+    {
+        Task<int> AddBillAsync(Bill bill);
+
+        Task<int> UpdateBillAsync(Bill bill);
+
+        Task<int> DeleteBillAsync(Bill bill);
```
Comment (JmlKhon) [inline TestHotel/TestHotel.Service/Service/IServices/IBillService.cs]: delete Id bo'yicha delete qimassmidi?

File: TestHotel/TestHotel.Service/Service/IServices/IBillService.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+using System.Threading.Tasks;
+using TestHotel.DataAccess.Model;
+
+namespace TestHotel.Service.Service.IServices
+{
+    internal interface IBillService
+    {
+        Task<int> AddBillAsync(Bill bill);
+
+        Task<int> UpdateBillAsync(Bill bill);
```
Comment (JmlKhon) [inline TestHotel/TestHotel.Service/Service/IServices/IBillService.cs]: Update ham Id bo'yicha update qimasmidi

File: TestHotel/TestHotel.Service/Service/Services/EmployeeService.cs (hunk 1 above)
Comment (isfandiyoro5) [reply TestHotel/TestHotel.Service/Service/Services/EmployeeService.cs]: done

File: TestHotel/TestHotel.Service/Service/Services/EmployeeService.cs (hunk 2 above)
Comment (isfandiyoro5) [reply TestHotel/TestHotel.Service/Service/Services/EmployeeService.cs]: Ok

File: TestHotel/TestHotel.Service/Service/Services/BillService.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            _billRepository = billRepository;
+        }
+        public async Task<int> AddBillAsync(Bill bill)
+        {
+            return await _billRepository.AddBillAsync(bill);
+        }
+
+        public async Task<int> UpdateBillAsync(Bill bill)
+        {
+            var billExist = await GetBillByIdAsync(bill.InvoiceNumber);
```
Comment (JmlKhon) [inline TestHotel/TestHotel.Service/Service/Services/BillService.cs]: change var name

File: TestHotel/TestHotel.Service/Service/Services/BillService.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            if (billExist != null)
```
Comment (JmlKhon) [inline TestHotel/TestHotel.Service/Service/Services/BillService.cs]: change to "is not"

File: TestHotel/TestHotel.Service/Service/Services/BillService.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            {
+                return await _billRepository.UpdateBillAsync(bill);
+            }
+            else
+            {
+                return 0;
```
Comment (JmlKhon) [inline TestHotel/TestHotel.Service/Service/Services/BillService.cs]: throw new exception with message

**Review by JmlKhon — CHANGES_REQUESTED** [review]

File: TestHotel/TestHotel.Service/Service/IServices/IBillService.cs (hunk 3 above)
Comment (isfandiyoro5) [reply TestHotel/TestHotel.Service/Service/IServices/IBillService.cs]: Done

File: TestHotel/TestHotel.Service/Service/IServices/IBillService.cs (hunk 4 above)
Comment (isfandiyoro5) [reply TestHotel/TestHotel.Service/Service/IServices/IBillService.cs]: Done

File: TestHotel/TestHotel.Service/Service/Services/BillService.cs (hunk 5 above)
Comment (isfandiyoro5) [reply TestHotel/TestHotel.Service/Service/Services/BillService.cs]: Done

File: TestHotel/TestHotel.Service/Service/Services/BillService.cs (hunk 6 above)
Comment (isfandiyoro5) [reply TestHotel/TestHotel.Service/Service/Services/BillService.cs]: Done

File: TestHotel/TestHotel.Service/Service/Services/BillService.cs (hunk 7 above)
Comment (isfandiyoro5) [reply TestHotel/TestHotel.Service/Service/Services/BillService.cs]: Done

**Review by SheraliErkayev — APPROVED** [review]

**Review by MirazamTUT — APPROVED** [review]

**Review by JmlKhon — APPROVED** [review]

**Review by mrIminjonov — APPROVED** [review]

**Pull request merged**